aws-cpp-sdk-transfer/source/transfer/UploadFileRequest.cpp

```diff
@@ -91,56 +90,125 @@ m_getObjectRetries(0),
 m_listObjectsRetries(0),
 m_headBucketRetries(0)
 {
-    if (m_fileStream.good() && m_fileStream.is_open())
+    if (fileName != "")

```

Comment:
I've now refactored the constructors to address this issue and improve the code organization.